registry/storage/driver/swift/swift_test.go

```diff
@@ -0,0 +1,154 @@
+package swift
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/ncw/swift/swifttest"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/testsuites"
+
+	"gopkg.in/check.v1"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { check.TestingT(t) }
+
+type SwiftDriverConstructor func(rootDirectory string) (*Driver, error)
+
+func init() {
+	var (
+		username           string
+		password           string
+		authURL            string
+		tenant             string
+		tenantID           string
+		domain             string
+		domainID           string
+		container          string
+		region             string
+		prefix             string
+		insecureSkipVerify bool
+		swiftServer        *swifttest.SwiftServer
+		err                error
+	)
+	if username = os.Getenv("OS_USERNAME"); username == "" {
```

Comment:
"OS" is probably not specific enough for this use case. Perhaps "OPENSTACK" or "SWIFT" would be better.